bothan-core/src/store.rs

```diff
@@ -1,90 +1,81 @@
 use std::collections::{HashMap, HashSet};
+use std::path::Path;
 
-use tokio::sync::{Mutex, RwLock};
+pub use manager::ManagerStore;
+use rust_rocksdb::{Options, DB};
+use serde::Serialize;
+use tokio::sync::RwLock;
+use tracing::info;
+pub use worker::WorkerStore;
 
+use crate::registry::Registry;
+use crate::store::errors::Error;
 use crate::types::AssetInfo;
-use crate::worker::AssetStatus;
 
-// TODO: Store should have namespace for different sources
+pub mod errors;
+mod manager;
+mod worker;
+
 pub struct Store {
     asset_store: RwLock<HashMap<String, AssetInfo>>,
```

Comment:
I think we should move all the lock to the level of whoever holding the reference to store and not each individual field. It will make things consistent/simpler for function that need to interact with store many times so it won't need to hold several logs (one for each field) concurrently.